Mathlib/Data/Fintype/Quotient.lean

```diff
@@ -134,6 +134,34 @@ lemma finLiftOn_mk (a : ∀ i, α i) :
   rw [finChoice_eq]
   rfl
 
+section
+
+variable {ι₁ : Type*} [Fintype ι₁] [DecidableEq ι₁] {ι₂ : Type*} [Fintype ι₂] [DecidableEq ι₂]
+  {α : ι₁ → Sort*} {S₁ : ∀ i, Setoid (α i)} {β : ι₂ → Sort*} {S₂ : ∀ i, Setoid (β i)} {φ : Sort*}
+  (q₁ : ∀ i, Quotient (S₁ i)) (q₂ : ∀ i, Quotient (S₂ i))
+
+/-- Lift a binary function from its finitely indexed types `∀ i : ι₁, α i`, `∀ i : ι₂, β i` to
+a binary function on quotients. This is analogous to the combination of `Quotient.finLiftOn`
+and `Quotient.liftOn₂`. -/
+def finLiftOn₂ (q₁ : ∀ i, Quotient (S₁ i)) (q₂ : ∀ i, Quotient (S₂ i))
+    (f : (∀ i, α i) → (∀ i, β i) → φ)
+    (c : ∀ (a₁ : ∀ i, α i) (b₁ : ∀ i, β i) (a₂ : ∀ i, α i) (b₂ : ∀ i, β i),
+        (∀ i, a₁ i ≈ a₂ i) → (∀ i, b₁ i ≈ b₂ i) → f a₁ b₁ = f a₂ b₂) :
```

Comment:
q1 and q2 are already in the variables, and f and c are also shared with the theorem below, so maybe you could just move them all to the variables?